tests/test_asyncio/test_connection.py

```diff
@@ -41,6 +42,50 @@ async def test_invalid_response(create_redis):
     await r.connection.disconnect()
 
 
+@pytest.mark.onlynoncluster
```

Comment:
this test is a bit overboard perhaps

Essentially, I couldn't find a reliable way to validate the locking does what we want without patching the connection and call_with_retry objects.

Let me know if you think this is unnecessary, or perhaps we can improve this!